src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import type {ImageStyle} from 'react-native'
+import {
+  Image,
+  ImageResizeMode,
+  ImageSourcePropType,
+  StyleProp,
+  StyleSheet,
+  TouchableOpacity,
+  ViewStyle,
+  Insets,
+} from 'react-native'
+import {metrics, hitSlop as defaultHitSlop} from '../../helpers/metrics'
+
+export type IconProps = {
+  source: ImageSourcePropType
+  size?: number
+  disabled?: boolean
+  color?: string
+  hitSlop?: Insets
+  style?: StyleProp<ImageStyle>
+  resizeMode?: ImageResizeMode
+  testID?: string
+  onPress?: () => void
+  onLongPress?: () => void
+  buttonStyle?: StyleProp<ViewStyle>
+}
+
+export const Icon: React.FunctionComponent<IconProps> = ({
+  source,
+  size = metrics.medium,
+  disabled = false,
+  color,
+  hitSlop = defaultHitSlop,
+  style,
+  resizeMode = 'contain',
+  testID,
+  onPress,
+  onLongPress,
+  buttonStyle,
+}) => (
+  <TouchableOpacity
+    testID={testID}
+    disabled={(!onPress && !onLongPress) || disabled}
+    onPress={onPress}
+    onLongPress={onLongPress}
+    style={buttonStyle}
+    hitSlop={hitSlop}>
+    <Image
+      testID="icon-image"
+      source={source}
+      style={[
+        {
+          width: size,
+          height: size,
+          tintColor: color,
+        },
+        StyleSheet.flatten(style),
+      ]}
+      resizeMode={resizeMode || 'contain'}
```

Comment:
We already have default value in line 36

Reply:
Done